core/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java

```diff
@@ -173,7 +173,7 @@ public void handleResponse(final Response response) {
 
                     @Override
                     public void handleException(TransportException exp) {
-                        perform(exp);
+                        listener.onFailure(exp);
```

Comment:
@jimferenczi Thanks for your comment. However ,I'm not sure what you say... I thought we have already the null check of shardIt , https://github.com/johtani/elasticsearch/blob/fix/no_respond_analyze_api_without_index/core/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java#L156 .
Then I thought perform method always fail.
Is my understanding wrong? Do I overlook what it is?